lib/commands/createcache.js

```diff
@@ -62,7 +74,25 @@ function createCache(opts, request, done){
         "persistent" : true,
         "skipCacheIfElementSizeInKBExceeds" : 512
       }
-
+  
+  if(opts.cacheExpiryByDate){
+    var expirySettings = {
+      "expiryDate" : {
+        "value" : opts.cacheExpiryByDate
+      },
+      "valuesNull" : false
+    }
+    createCachePayload.expirySettings = expirySettings;
+  } else if(opts.cacheExpiryInSecs){
+    var expirySettings = {
+      "timeoutInSec": {
+        "value": opts.cacheExpiryInSecs
+      },
+      "valuesNull": false
+    }
+    createCachePayload.expirySettings = expirySettings;
+  }     
```

Comment:
The same feedback above, but I think this one is actually broken in that it would replace the `expiryDate` if you provided both `cacheExpiryByDate` and `cacheExpiryInSecs`.

Reply:
If both `cacheExpiryByDate` and `cacheExpiryInSecs` are provided then only expiryDate will be passed in the payload. Should we add a note in readme file?
If only `cacheExpiryInSecs` is provided then block of code will overwrite expiryDate setting with timeoutInSec setting.

Either i have to use current code or use below code. I will prefer existing code. Your comments?
`createCachePayload.expirySettings = {} //Clear the default values,
 createCachePayload.expirySettings.timeoutInSec = {}
 createCachePayload.expirySettings.timeoutInSec.value = opts.cacheExpiryInSecs
 createCachePayload.expirySettings.valuesNull =  false`

Reply:
So it looks like `cacheExpiryByDate` and `cacheExpiryInSecs` are exclusive?

Reply:
If both `cacheExpiryByDate` and `cacheExpiryInSecs` are passed to apigee api it accepts the request and only `cacheExpiryInSecs` is stored. No exclusion errors are returned by apigee api.

Do we want to pass both the values or only pass customer provided value while making apigee api?

Reply:
If the customer _(CLI consumer)_ provides both, I say we should provide both values to the Edge MGMT API and let Edge decide what to do.  That's why I was confused in the first place because it seemed like we were allowing them to provide both, but only `cacheExpiryInSecs` was being sent because we would overwrite the value of `createCachePayload.expirySettings`.

Reply:
PR is updated to pass both `cacheExpiryByDate` and `cacheExpiryInSecs` to Edge MGMT API.

Summary for different scenarios
- No expiry settings are provided by customer (CLI consumer), then default value for `cacheExpiryByDate` (default date 12-31-9999) will be sent to Edge MGMT API. Edge MGMT API stores `cacheExpiryByDate`.

- Customer (CLI consumer) provides `cacheExpiryByDate`, then only `cacheExpiryByDate` (user input) will be send to Edge MGMT API. Edge MGMT API stores `cacheExpiryByDate`.

- Customer (CLI consumer) provides `cacheExpiryInSecs`, then both `cacheExpiryByDate` (default date 12-31-9999) and `cacheExpiryInSecs` (user input) will be send to Edge MGMT API. Edge MGMT API stores `cacheExpiryInSecs` only.

- Customer (CLI consumer) provides `cacheExpiryByDate` and `cacheExpiryInSecs`, then both `cacheExpiryByDate` (user input) and `cacheExpiryInSecs` (user input) will be send to Edge MGMT API. Edge MGMT API stores `cacheExpiryInSecs` only.

Pl review and provide your feedback.

Reply:
@whitlockjc could you review?